implementations/code_df/open_issue_age.py

```diff
@@ -0,0 +1,138 @@
+from datetime import datetime
+
+import pandas as pd
+import numpy as np
+
+import utils
+from issue import Issue
+
+
+class OpenIssueAge(Issue):
+
+    def __init__(self, data_list, date_range=(None, None)):
+        """
+        Class for Age of Open Issues
+
+        :param data_list: A list of dictionaries, each element a line from the
+            JSON file
+        :param date_range: A tuple which represents the start and end date of
+            interest
+        """
+        super().__init__(data_list, date_range)
+
+        self.df = self._add_open_issue_age_col(self.df)
+
+    def compute(self):
+        """
+        Compute the average open issue age for all issues in the Perceval data.
+
+        :returns avg_open_issue_ages: the average age of open
+            issues
+        """
+
+        open_issue_ages = [age for age in self.df['open_issue_age']
+                           if not pd.isna(age)]
+
+        avg_open_issue_ages = np.mean(open_issue_ages)
+        return avg_open_issue_ages
+
+    def compute_timeseries(self, period="month"):
+        """
+        The metric value is computed for each fixed interval of time
+        from the "since" date to the "until" date arguments, specified
+        during object initiation.
+        The fixed time interval can be either a month or a week.
+
+        :param period: A string which can be either "month" or "week".
+
+        :returns dataframe: A DataFrame whose rows each represent an interval
+            of "period" and the count for that interval.
+        """
+
+        if self.since is None:
+            self.since = utils.get_date(self.df, 'since')
+        if self.until is None:
+            self.until = utils.get_date(self.df, 'until')
+
+        df = self.df
+        if period == "month":
```

Comment:
The code under `if` and `else` is almost the same. You could have a dictionary such as
```
period_frequency_map = {"month": "M", "week": "W", ....}
```

and use the `period` variable to determine what the frequency should be. What do you think @Polaris000?

Reply:
Good point @aswanipranjal! @jgbarah probably noticed this issue and told me to pass the pandas timescale values like `'M', 'W',` etc directly to the parameter `period` instead "month", "week", etc. 

In any case, I am re-implementing `compute_time_series` using `DataFrame.resample(..)`